state/store.go

```diff
@@ -232,7 +256,7 @@ func saveValidatorsInfo(db dbm.DB, height, lastHeightChanged int64, valSet *type
 	valInfo := &ValidatorsInfo{
 		LastHeightChanged: lastHeightChanged,
 	}
-	if lastHeightChanged == height {
+	if height == lastHeightChanged || height%valSetCheckpointInterval == 0 {
```

Comment:
```go
func valSetChangedOrIsCheckpointHeight(height) bool {
	return height == lastHeightChanged || height%valSetCheckpointInterval == 0
}
```

Reply:
Yeah, I also thought about introducing vars to make this more readable. BUt thought it is OK as it is. 

Either the function you proposed or sth along the lines of:
```go
 hasValSetChanged := height == lastHeightChanged 
 isCheckPointHeight := height%valSetCheckpointInterval == 0
 if hasValSetChanged || isCheckPointHeight {
    // ...
 }
```
I prefer the vars instead of the function but both are fine.